blockchain/types/transaction.go

```diff
@@ -300,16 +300,16 @@ func (tx *Transaction) EffectiveGasTip(baseFee *big.Int) *big.Int {
 // After hard fork, we need to consider new getEffectiveGasPrice API that operates differently
 // effectiveGasPrice = msg.EffectiveGasPrice(baseFee)
 func (tx *Transaction) EffectiveGasPrice(baseFee *big.Int) *big.Int {
-	if tx.Type() == TxTypeEthereumDynamicFee {
-		if baseFee != nil {
-			return baseFee
+	if baseFee != nil {
+		return baseFee
+	} else {
+		if tx.Type() == TxTypeEthereumDynamicFee {
+			baseFee = new(big.Int).SetUint64(params.ZeroBaseFee)
+			te := tx.GetTxInternalData().(TxInternalDataBaseFee)
+			return math.BigMin(new(big.Int).Add(te.GetGasTipCap(), baseFee), te.GetGasFeeCap())
```

Comment:
Along with our policy, i think we need to return `baseFee` after hard fork for `TxTypeEthereumDynamicFee` type.
If `baseFee` is `250Ston`, `GasFeeCap` is `500Ston`(bigger than baseFee), and `GasTipCap` is  `100Ston`, then this logic will return 350Ston. But we need to return `250Ston`.

We can keep this logic for before hard fork(when baseFee is 0).

Reply:
yes, right